src/transactions/SetOptionsOpFrame.cpp

```diff
@@ -18,9 +18,11 @@ namespace stellar
 {
 
 static const uint32 allAccountFlags =
-    (AUTH_REQUIRED_FLAG | AUTH_REVOCABLE_FLAG | AUTH_IMMUTABLE_FLAG);
+    (AUTH_REQUIRED_FLAG | AUTH_REVOCABLE_FLAG | AUTH_IMMUTABLE_FLAG |
```

Comment:
After reviewing the way that `trustLineFlagIsValid` was done in the next commit, I'm also questioning whether we should be using a similar approach for the `MASK_ACCOUNT_FLAGS`. I think it would probably be slightly better in that the invariant would be able to check for flags that were introduced in later versions when running earlier versions, which is valuable for ensuring correctness in parallel catchup. What do you think?

Reply:
but if you mean that there should be a `accountFlagIsValid` equivalent to `trustLineFlagIsValid` that we can use both invariants and in `setOptions` that makes sense to me

Reply:
The last commit (bf6bf6e068b7ad24dc3d7418c444cd89b04a5c3e) adds the account flag validity check to `LedgerEntryIsValid`, but it works a little different from trustline flag validation.

We can reuse `trustLineFlagIsValid` in the invariants and `AllowTrustOpFrame::doCheckValid` because `mAllowTrust.authorize` is the next expected state of the trustline flags. ` SetOptionsOpFrame` doesn't work the same way because its parameters change specific bits in the flag. Due to this, we need three different checks - 
1. At validation - check that parameter bits are valid. 
2. At apply - check that the revocable/clawback combo is valid.
3. Invariants - check that the both conditions above are true.

I kept these changes as a separate commit for now, but will squash it if we think this is worth doing. We will also have a very similar issue with `SetTrustlineFlagsOp`, so the pattern used here should be used in the new operation.

Reply:
The cap will also need to be updated to reflect `MASK_ACCOUNT_FLAGS_V16`.